### ISSUE TITLE
font rendering

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#5702]
This is more of a meta ticket since there are several problems here. We needed to remove polyfonts because of the license. There seems to be no alternative library which renders text as polygons so we need to use bitmaps. We decided not to use [freetype](https://www.freetype.org/index.html) because of the sheer size and the two step process involved of first generating the bitmap and then including it but we may need to revisit that decision. Our current decision [fontstash](https://github.com/memononen/fontstash/) does everything "live" but is no longer maintained (and its current OpenGL3 implementation does not work so we are stuck with older OpenGL code here). Currently visible problems mainly concern font rendering which does not happen on the main screen, e.g. black boxes appear on
- [ ] Xvfb
- [ ] remote desktops via RDP
- [ ] gl2ps output

**Comment by behrisch** [comment]
Also a good read in this context: https://www.opengl.org/archives/resources/faq/technical/fonts.htm

**Comment by palvarezlopez** [comment]
@behrisch we can implement our own font using the current implemented drawGL(...) functions. We only need to create a function that has a string and a position as inputs:

1) Separate string in n chars: test -> 4, 
2) every char is drawn in a "rectangle": "test" -> [t] [e] [s] [t]
3) calculate position of  middle point: [t] [e] * [s] [t]
4) drawn every char using an already implemented function for every char

**Comment by behrisch** [comment]
@palvarezlopez I don't think it is an option. Eventually we want to support unicode. But if you find a tool which can translate a font into gl commands I would be happy to use it.

**Comment by palvarezlopez** [comment]
We cannot use FreeType due FOX Toolkit :(

**Comment by namdre** [comment]
Actually, there is nothing that prevents FOX Toolkit from using shaders. It's just that we're targeting an older OpenGL Version to ensure compatibility on Windows (without worring about graphics drivers). Shaders are even mentioned on the front page of http://fox-toolkit.org/. 
One solution to the portability issue might be https://en.wikipedia.org/wiki/ANGLE_(software) which aims to allow for OpenGL compatibility on Windows and Mac.
